integrations/operator/embeddedtbot/bot.go

```diff
@@ -0,0 +1,156 @@
+package embeddedtbot
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/lib/tbot"
+	"github.com/gravitational/teleport/lib/tbot/config"
+)
+
+// EmbeddedBot is an embedded tBot instance to renew the operator certificates.
+type EmbeddedBot struct {
+	cfg *config.BotConfig
+
+	credential *config.UnstableClientCredentialOutput
+
+	// mutex protects started, cancelCtx and errCh
+	mutex     sync.Mutex
+	started   bool
+	cancelCtx func()
+	errCh     chan error
+}
+
+// New creates a new EmbeddedBot from a BotConfig.
+func New(botConfig *BotConfig) (*EmbeddedBot, error) {
+	credential := &config.UnstableClientCredentialOutput{}
+
+	cfg := (*config.BotConfig)(botConfig)
```

Comment:
We should be careful about using this to copy - as some of the fields within the Bot may be pointers - I don't think this is a problem in the current usage though.